### ISSUE TITLE
[filesys] collecting /proc/*/mountinfo raises AVC denials

### ISSUE THREAD

**Issue by pmoravec** [sosreport/sos#1399]
When sosreport attempts to collect `/proc/[0-9]*/mountinfo` via `filesys` plugin (enabled by default) on Enforcing SELinux system, below AVCs are raised:

    type=AVC msg=audit(1534763891.040:3455): avc:  denied  { relabelto } for  pid=12759 comm="sosreport" name="mountinfo" dev="dm-0" ino=15039276 scontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tcontext=system_u:system_r:init_t:s0 tclass=file permissive=0

It is due to the fact sosreport collects any file also _with_ its SELinux context, and the relabel fails.

The files are collected by sosreport, just with wrong label (that shouldnt matter here).

I understand sos should not be allowed to label files to context like e.g. `system_u:system_r:init_t:s0` while the `mountinfo` files can be interesting (they were added in #1212).

There are 3 options:

- leave the harmless AVCs (as we are not interested in the context that relabel fails) - but sosreport shouldnt raise AVCs
- dont collect those files
- collect them without preserving permissions incl. SElinux context (needs new option of `add_copy_spec`)

I personally vote for the 3rd option as the feature makes sense - if we will be interested in files content but not their permissions.

@bmr-cymru , @TurboTurtle : any thoughts?

**Comment by TurboTurtle** [comment]
I agree with option 3. Particularly because these files were requested by our filesystem support team and were mentioned to be "extremely valuable", so I don't think option 2 should be considered even if we don't go with 3.

**Comment by bmr-cymru** [comment]
Simple blacklist (global or per-policy: some discussion going on on IRC now): if it's in `/proc` or `/sys` we need to avoid using `shutil.copystat()` (which attempts to copy xattrs, resulting in AVCs when the containing file system is forbidden by policy from containing that label).

We'll need to grow our own code to copy the remaining file properties (mode, times etc.), but this shouldn't be a big deal (we've been ignoring the exceptions that these errors generate in `sos.archive` for years).
